qemu/tests/block_with_write_threshold.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import re
+
+from avocado.utils.wait import wait_for
+
+from virttest import data_dir
+from virttest import error_context
+from virttest import utils_disk
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.qemu_storage import QemuImg
+
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test the block write threshold for block devices.
+    Steps:
+        1. Create a data disk.
+        2. Boot up the guest with a data disk attached.
+        3. Set block write threshold for the data block drive in QMP.
+        4. Login to guest then do stress io to trigger the threshold.
+        5. Verify the event 'BLOCK_WRITE_THRESHOLD' in QMP.
+        6. Set block write threshold to 0 for the data block drive in QMP
+           which will not trigger the threshold.
+        7. Login to guest then do stress io.
+        8. Verify event 'BLOCK_WRITE_THRESHOLD' in QMP which should not be
+           triggered.
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def get_node_name(image_tag):
+        """ Get the node name. """
+        img_param = params.object_params(image_tag)
+        file = QemuImg(img_param, data_dir.get_data_dir(), image_tag).image_filename
```

Comment:
better to use `storage.get_image_filename`

Reply:
Thanks, have updated, pls help me to review it again.